pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -184,11 +263,48 @@ func (c *Reconciler) updateStatus(taskrun *v1alpha1.TaskRun) (*v1alpha1.TaskRun,
 	if err != nil {
 		return nil, err
 	}
-	if !reflect.DeepEqual(newtaskrun.Status, taskrun.Status) {
+	if !reflect.DeepEqual(taskrun.Status, taskrun.Status) {
 		newtaskrun.Status = taskrun.Status
 		// TODO: for CRD there's no updatestatus, so use normal update
 		return c.PipelineClientSet.PipelineV1alpha1().TaskRuns(taskrun.Namespace).Update(newtaskrun)
 		//	return configClient.UpdateStatus(newtaskrun)
 	}
 	return newtaskrun, nil
 }
+
+func (c *Reconciler) getBuild(namespace, name string) (*buildv1alpha1.Build, error) {
+	return c.buildLister.Builds(namespace).Get(name)
+}
+
+func (c *Reconciler) createBuild(namespace string, b *buildv1alpha1.Build) (*buildv1alpha1.Build, error) {
+	return c.BuildClientSet.BuildV1alpha1().Builds(namespace).Create(b)
+}
+
+func (c *Reconciler) deleteBuild(namespace, name string) error {
+	return c.BuildClientSet.BuildV1alpha1().Builds(namespace).Delete(name, &metav1.DeleteOptions{})
+}
+
+func (c *Reconciler) deleteTaskRun(namespace, name string) error {
+	return c.PipelineClientSet.PipelineV1alpha1().TaskRuns(namespace).Delete(name, &metav1.DeleteOptions{})
+}
+
+// makeBuild creates a build from the task, using the task's buildspec.
+func makeBuild(t v1alpha1.Task) (*buildv1alpha1.Build, error) {
+	if t.Spec.BuildSpec == nil {
+		return nil, fmt.Errorf("nil BuildSpec")
+	}
+	return &buildv1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name,
```

Comment:
probably want this to be the name of the taskrun, since one task could have many task runs

Reply:
done